dataprofiler/tests/labelers/test_base_model.py

```diff
@@ -96,25 +96,99 @@ def test_set_parameters(self):
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2}, mock_model._label_mapping)
+
+        # assert successful add w/ same_as
+        label = 'NEW_LABEL_3'
+        base_model.BaseModel.add_label(mock_model, label, same_as='NEW_LABEL')
+        self.assertDictEqual(
+            {'NEW_LABEL': 1, 'NEW_LABEL_2': 2, 'NEW_LABEL_3': 1},
+            mock_model._label_mapping)
+
     def test_set_label_mapping_parameters(self):
 
         # setup mock
         mock_model = mock.Mock(spec=base_model.BaseModel)
+        mock_model._convert_labels_to_label_mapping.side_effect = \
+            base_model.BaseModel._convert_labels_to_label_mapping
 
         # base case
-        with self.assertRaisesRegex(ValueError,
-                                    "`label_mapping` must be a dict which maps "
-                                    "labels to index encodings."):
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
+            base_model.BaseModel.set_label_mapping(mock_model, label_mapping=None)
+
+        # base case
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
             base_model.BaseModel.set_label_mapping(
                 mock_model, label_mapping=None)
 
         # assert error for empty label_mapping dict
-        with self.assertRaisesRegex(ValueError,
-                                    "`label_mapping` must be a dict which maps "
-                                    "labels to index encodings."):
+        with self.assertRaisesRegex(TypeError,
+                                    "Labels must either be an encoding dict "
+                                    "which maps labels to index encodings or a "
+                                    "list."):
             base_model.BaseModel.set_label_mapping(mock_model, label_mapping={})
 
         # assert label_map set
         base_model.BaseModel.set_label_mapping(
             mock_model, label_mapping={'test': 'test'})
         self.assertDictEqual({'test': 'test'}, mock_model._label_mapping)
+
+    def test_convert_labels_to_encodings(self, *mocks):
+
+        # test label list to label_mapping
+        labels = ['a', 'b', 'd', 'c']
+        label_mapping = base_model.BaseModel._convert_labels_to_label_mapping(
+            labels, requires_zero_mapping=True)
+        self.assertDictEqual(dict(a=0, b=1, d=2, c=3), label_mapping)
+
+        # test label dict to label_mapping
+        labels = dict(a=1, b=2, d=3, c=4)
+        label_mapping = base_model.BaseModel._convert_labels_to_label_mapping(
+            labels, requires_zero_mapping=True)
```

Comment:
This may bring up a design question, but should requires_zero_mapping only be used/specified if the labels are given as a list? Should we raise a warning or anything in this case or is it on the user to understand that it only affects the mapping generated from a list.

Reply:
the labels are given as a dict, it is assumed the user is intentionally assigning the mapping. if they are provided as a list, the user is leaving the preference up to the model. In this case, the user doesn't really need to know what's happening below IMO since they have given the model autonomy to set it up how it needs it.

I welcome further thoughts.

Reply:
I understand what it means in the context of providing a list, but I am saying that requires_zero_mapping doesn't do anything when given a dict, so I just think there should be some reflection that of that in the docstring or a warning raised if a dict is passed and requires_zero_mapping is specified as non-None, since doing so implies the user doesn't understand what/when/how that parameter is used.

Reply:
Warnings / errors are provided to the user for the case of our character model if the user inputs an improper dict with directions of how to fix.

Reply:
These errors are also model dependent, so I'm not sure there's a generic error that could be made.

In addition, with these new changes, it might be that `requires_zero_mapping` is tech debt and can be removed as the model themselves will alter the list / dict arrangements rather than a base class or the data labeler